packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -137,26 +139,37 @@ export async function runHttpQuery(
         extensions &&
         extensions.persistedQuery
       ) {
-        // It looks like we've received an Apollo Persisted Query. Apollo Server
-        // does not support persisted queries out of the box, so we should fail
-        // fast with a clear error saying that we don't support APQs. (A future
-        // version of Apollo Server may support APQs directly.)
-        throw new HttpQueryError(
-          // Return 200 to simplify processing: we want this to be intepreted by
-          // the client as data worth interpreting, not an error.
-          200,
-          JSON.stringify({
-            errors: [
-              {
-                message: 'PersistedQueryNotSupported',
-              },
-            ],
-          }),
-          true,
-          {
-            'Content-Type': 'application/json',
-          },
-        );
+        // It looks like we've received an Apollo Persisted Query.
+        const sha = extensions.persistedQuery.sha256Hash;
+
+        queryString = await optionsObject.cache.get(sha);
```

Comment:
Guard against cache not existing (you should be allowed to explicitly disable PQ), send the PersistedQueryNotSupported error in that case.

If you really are encouraging people to pass literally the same cache to multiple features, we should prefix cache keys with some identifier character so that different features don't interact. On the other hand probably there's no good reason to pass the same in-memory hash to multiple features, and for memcache maybe it's just the job of the person building the memcache object to specify a per-feature prefix?

The Engine trace proto has persisted_query_hit and persisted_query_register bools (we don't track PQ misses since there's no trace, and also neither of these are currently shown in the UI).  It would be great to get equivalent bools into, say, requestDidStart.